From a5ba41cefffbd6ce17668d5271d2597c8c77a5e5 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Fri, 23 Aug 2019 15:59:43 +0800
Subject: [PATCH] Revert "clk: rockchip: Modify uart frac divider rule"

This reverts commit aa34d0fe717d26c298595b083458099bc27e4560.
This rule does not cover the parent priority selection problem.

Change-Id: Id75d46548e28da378420f614b45a6b71c6601843
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 drivers/clk/rockchip/clk.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/clk/rockchip/clk.c b/drivers/clk/rockchip/clk.c
index 7a9c3929a333..f4f716f77b07 100644
--- a/drivers/clk/rockchip/clk.c
+++ b/drivers/clk/rockchip/clk.c
@@ -195,19 +195,6 @@ static void rockchip_fractional_approximation(struct clk_hw *hw,
 	u32 div;
 
 	p_rate = clk_hw_get_rate(clk_hw_get_parent(hw));
-
-	if (strstr(clk_hw_get_name(hw), "uart")) {
-		if (rate <= 24000000) {
-			*parent_rate = 24000000;
-		} else {
-			if (fd->max_prate)
-				*parent_rate = fd->max_prate;
-			else
-				*parent_rate = 480000000;
-		}
-		goto frac_ration;
-	}
-
 	if (((rate * 20 > p_rate) && (p_rate % rate != 0)) ||
 	    (fd->max_prate && fd->max_prate < p_rate)) {
 		p_parent = clk_hw_get_parent(clk_hw_get_parent(hw));
@@ -232,7 +219,6 @@ static void rockchip_fractional_approximation(struct clk_hw *hw,
 		}
 	}
 
-frac_ration:
 	/*
 	 * Get rate closer to *parent_rate to guarantee there is no overflow
 	 * for m and n. In the result it will be the nearest rate left shifted
-- 
2.35.3

